README.md

```diff
@@ -1,14 +1,57 @@
-# minGPT Lightning
+# minGPT with Lightning & DeepSpeed
 
 ![mingpt](mingpt.jpg)
 
 [![Open in Streamlit](https://static.streamlit.io/badges/streamlit_badge_black_white.svg)](https://share.streamlit.io/seannaren/mingpt/streamlit/app.py)
 
-Modified [Andrej's](https://github.com/karpathy/minGPT) and [William's](https://github.com/williamFalcon/minGPT) awesome code to provide a benchmark script and show how to train a model from scratch at scale using Lightning.
+Modified [Andrej's](https://github.com/karpathy/minGPT) and [William's](https://github.com/williamFalcon/minGPT) awesome code to provide a minimal example of how to pair Lightning and DeepSpeed with a minimal GPT model.
 
-- [DeepSpeed Stage 3 Example](https://github.com/SeanNaren/minGPT/tree/deepspeed)
-- [FairScale Fully Sharded Example](https://github.com/SeanNaren/minGPT/tree/fairscale)
-- [Benchmark Suite (Fairscale and DeepSpeed capabilities) + Streamlit Visualization](https://github.com/SeanNaren/minGPT/tree/benchmark)
+*Note: this minimal example won't be as efficient/optimized as other specialized repos due to keeping it minimal and readable, but large model training is still achievable.* 
+
+## Usage
+
+```
+pip install -r requirements.txt
+```
+
+### Training Billion+ Parameter GPT Models
+
+A lot of information has been taken from the very helpful [Lightning Model Parallel Documentation](https://pytorch-lightning.readthedocs.io/en/latest/advanced/model_parallel.html#fully-sharded-training).
+
+In the below examples batch size is set to 1 to try reduce VRAM as much as possible, but you can scale that with your compute. In the below case we could scale the batch size significantly to fill the left over GPU memory.
+
+For 20B/45B parameter models, you'll need a reasonable amount of CPU RAM as we offload partitions to the CPU. For the 45B parameter model, you'll need around 1TB of CPU memory which is the default for the p4d.24xlarge instance in AWS (roughly 9 dollars an hour for a spot instance).
+
+Note that we enable offloading for ~20B+ param sizes. Offloading has a huge impact on throughput and in most cases when training from scratch should be turned off. You should consider scaling the number of GPUs rather than enabling offloading at these model sizes.
```

Comment:
```suggestion
Note that we enable CPU offloading. Offloading has a huge impact on throughput and in most cases when training from scratch should be turned off. You should consider scaling the number of GPUs rather than enabling offloading at these model sizes.
```